src/main/java/ch/njol/skript/lang/VariableString.java

```diff
@@ -345,10 +346,22 @@ private static void checkVariableConflicts(final String name, final StringMode m
 			pattern = Pattern.compile(Pattern.quote(name));
 		}
 		if (!SkriptConfig.disableVariableConflictWarnings.value()) {
-			for (final Entry<String, Pattern> e : variableNames.entrySet()) {
-				if (e.getValue().matcher(name).matches() || pattern.matcher(e.getKey()).matches()) {
-					Skript.warning("Possible name conflict of variables {" + name + "} and {" + e.getKey() + "} (there might be more conflicts).");
-					break;
+			Config cs = ScriptLoader.currentScript; //Eclipse's nullness forced me to do this
+			if (cs != null) {

```

Comment:
You could do `assert cs != null;` instead of the `if` when you're 100% sure something should be never null

Reply:
currentScript can be null when skript is not parsing syntax from a file. Therefore no assert :3

Reply:
No, the `cs.getFile()` can be null in this situation but the Config is never null as far as I can tell, that would be weird at least

Reply:
it absolutely can be null, for example running Effect#parse when it is null

Reply:
https://github.com/SkriptLang/Skript/blob/a243fc44abdac92cef39302ec7958b66d23eca3c/src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java#L57 then this is wrong